resourseleak/TestConfidenceScores.java

```diff
@@ -0,0 +1,114 @@
+import com.amazon.arest.client.aaa.AaaFeature;
+import com.amazon.coral.metrics.Metrics;
+import com.amazonaws.AmazonServiceException;
+import com.amazonaws.a4b.endpointoperationservice.dependency.accessor.MediaStorageServiceAccessor;
+import com.amazonaws.a4b.endpointoperationservice.dependency.exception.DependencyServiceException;
+import com.amazonaws.a4b.endpointoperationservice.logic.OperationLogic;
+import com.amazonaws.a4b.endpointoperationservice.logic.S3Logic;
+import com.amazonaws.a4b.endpointoperationservice.model.OperationModel;
+import com.amazonaws.a4b.endpointoperationservice.stepfunctions.model.UploadMediaActivityInput;
+import com.amazonaws.a4b.endpointoperationservice.util.SerializationUtils;
+import com.amazonaws.a4b.platform.common.exception.HttpClientException;
+import com.amazonaws.a4b.platform.exception.InvalidDeviceStateException;
+import com.amazonaws.a4b.platform.utils.HttpClientHelper;
+import com.amazonaws.services.s3.model.S3Object;
+import org.apache.commons.io.IOUtils;
+
+import javax.ws.rs.client.WebTarget;
+import javax.ws.rs.core.Response;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.StringWriter;
+import java.time.Clock;
+
+public class TestConfidenceScores {
+
+    private static final int CONNECTION_TIME_OUT = 3000;
+    private static final String APP_NAME = "A4BAnalyticsLambda";
+    private final MediaStorageServiceAccessor mssAccessor;
+    private final S3Logic s3Logic;
+    private final OperationLogic operationLogic;
+    private final Clock clock;
+
+
+    public String testConfidenceScores1() throws IOException {
+        final StringWriter writer = new StringWriter();
+        IOUtils.copy(this.getClass().getResourceAsStream("/report_template.html"), writer);
+        return writer.toString();
+    }
+
+    public RegisterAVSDeviceResponse testConfidenceScores2(final String clientId, final String customerId, final String productId,
+                                                       final String productSerialNumber) {
+
+        try {
+            final String adrsRegisterAVSDeviceEndpoint = adrsEndpoint + ADRS_REGISTER_AVS_DEVICE_PATH;
+
+            final WebTarget target = client.target(adrsRegisterAVSDeviceEndpoint).register(AaaFeature.class).queryParam("clientId", clientId)
+                    .queryParam("accountPool", accountPool).queryParam("productSerialNumber", productSerialNumber).queryParam("customerId", customerId)
+                    .queryParam("productId", productId);
+            final Response response = target.request().property(SERVICE_KEY, SERVICE_NAME).property(OPERATION_KEY, OPERATION_NAME).get();
+
+            HttpClientHelper.inspectStatusCode(response, SERVICE_NAME);
+            return response.readEntity(RegisterAVSDeviceResponse.class);
+        } catch (HttpClientException e) {
+            throw new InvalidDeviceStateException(ADRS_ERROR_MESSAGE, e);
+        } catch (DependencyServiceException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: An exception is caught and re-thrown within the catch clause without any other operations being performed on the exception, making the catch clause unnecessary.

**Fix**: Perform additional operations on the exception that has been caught. You can wrap a standard exception into a custom exception to add additional information or special handling. When you do that, make sure to set the original exception as the cause. Otherwise, you will lose the stack trace and message of the original exception, which will make it difficult to analyze the event that caused the exception.  
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)